internal/domain/review_suggestion.go

```diff
@@ -0,0 +1,27 @@
+package domain
+
+import (
+	"github.com/invopop/jsonschema"
+)
+
+type ReviewSuggestion struct {
+	Comment  string `json:"comment"`
+	Position int    `json:"position"`
+}
+
+type ReviewSuggestionsResponse struct {
+	ReviewSuggestions []ReviewSuggestion `json:"review_suggestions"`
```

Comment:
There is no error handling or validation of the type `T` in `GenerateSchema`. If `T` does not contain any fields or is an invalid type for schema generation, it's unclear how this will behave. Consider adding checks or comments explaining the expected type constraints.